### DISCUSSION TITLE
Chromebook Can't Find Url for YouTube Video

### DISCUSSION THREAD

**Discussion by johnaweiss** [smplayer-dev/smplayer#779] [Q&A]
Hello, i installed SMTube and SMPlayer on Chromebook Linux. 
I'm able to run both applications, no problem. 

But if i paste a youtube url into SMPlayer, i get 

> "it wasn't possible to find the url for this video, Maybe you need to update youtube-dl". 

If i click "Yes" to update youtube-dl, i get
> "An error happened while downloading the file. Server unavailable."

I attempted to install from terminal with
`sudo apt install youtube-dl`

but i get
> "youtube-dl is already the newest version (2021.06.06-1)"

Any fix?
Thx

**Comment by smplayer-dev** [comment]
Select `yt-dlp` in preferences -> network instead of `youtube-dl`and try again.
If it still doesn't work, you can download yt-dlp from https://github.com/yt-dlp/yt-dlp

**Reply by johnaweiss** [reply]
Thx!

But I don't know which file to download, or what to do with it. 
I'm on Debian (Chromebook Linux).

Why can't i do `sudo apt-get install yt-dlp`?  I tried it, and seemed to install correctly. But didn't fix the problem. 

My prefs look like this (looked same before and after installing yt-dlp with above `apt-get` command):
![image](https://github.com/smplayer-dev/smplayer/assets/53209681/3c3bb2aa-245b-40de-aff7-7501f0959e89)

**Comment by johnaweiss** [comment]
Does this help solve it?

I changed Network prefs to
mpv + youtube-dl

Now if i try to load a url, i see "Starting...", but then i get:

```
/usr/bin/mpv --no-config --no-quiet --terminal --no-msg-color --input-ipc-server=/tmp/smplayer-mpv-1610 --msg-level=ffmpeg/demuxer=error --no-fs --hwdec=no --sub-auto=fuzzy --no-input-default-bindings --input-vo-keyboard=no --no-input-cursor --cursor-autohide=no --no-keepaspect --wid=4194651 --monitorpixelaspect=1 --osd-level=1 --osd-scale=1 --osd-bar-align-y=0.6 --sub-ass --embeddedfonts --sub-ass-line-spacing=0 --sub-scale=1 --sub-font=Arial --sub-color=#ffffffff --sub-shadow-color=#ff000000 --sub-border-color=#ff000000 --sub-border-size=0.75 --sub-shadow-offset=2.5 --sub-font-size=50 --sub-bold=no --sub-italic=no --sub-margin-y=8 --sub-margin-x=20 --sub-codepage=ISO-8859-1 --sub-pos=100 --volume=55 --cache=auto --screenshot-template=cap_%F_%p_%02n --screenshot-format=jpg --screenshot-directory=/tmp/smplayer_screenshots --audio-pitch-correction=yes --volume-max=110 --ytdl --term-playing-msg=MPV_VERSION=${=mpv-version:}
INFO_VIDEO_WIDTH=${=width}
INFO_VIDEO_HEIGHT=${=height}
INFO_VIDEO_ASPECT=${=video-params/aspect}
INFO_VIDEO_FPS=${=container-fps:${=fps}}
INFO_VIDEO_FORMAT=${=video-format}
INFO_VIDEO_CODEC=${=video-codec}
INFO_AUDIO_FORMAT=${=audio-codec-name}
INFO_AUDIO_CODEC=${=audio-codec}
INFO_AUDIO_RATE=${=audio-params/samplerate}
INFO_AUDIO_NCH=${=audio-params/channel-count}
INFO_LENGTH=${=duration:${=length}}
INFO_DEMUXER=${=current-demuxer:${=demuxer}}
INFO_SEEKABLE=${=seekable}
INFO_TITLES=${=disc-titles}
INFO_CHAPTERS=${=chapters}
INFO_TRACKS_COUNT=${=track-list/count}
METADATA_TITLE=${metadata/by-key/title:}
METADATA_ARTIST=${metadata/by-key/artist:}
METADATA_ALBUM=${metadata/by-key/album:}
METADATA_GENRE=${metadata/by-key/genre:}
METADATA_DATE=${metadata/by-key/date:}
METADATA_TRACK=${metadata/by-key/track:}
METADATA_COPYRIGHT=${metadata/by-key/copyright:}
INFO_MEDIA_TITLE=${=media-title:}
INFO_STREAM_PATH=${stream-path}
 --audio-client-name=SMPlayer --term-status-msg=STATUS: ${=time-pos} / ${=duration:${=length:0}} P: ${=pause} B: ${=paused-for-cache} I: ${=core-idle} VB: ${=video-bitrate:0} AB: ${=audio-bitrate:0} https://youtu.be/gWTnpy7sw2g

[ytdl_hook] ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output. 
[ytdl_hook] youtube-dl failed: unexpected error ocurred 
[ytdl_hook] It appears that your youtube-dl version is severely out of date. 
Failed to recognize file format.
Exiting... (Errors when loading file)
```

Any ideas?

Thx!

**Comment by smplayer-dev** [comment]
Download [yt-dlp_linux](https://github.com/yt-dlp/yt-dlp/releases/download/2023.07.06/yt-dlp_linux), rename it to yt-dlp, make it executable (chmod 755 yt-dlp) and copy to your `~/bin` folder (if it doesn't exit, create it).

**Reply by johnaweiss** [reply]
Thx! 
Still failed, when i pasted youtube url into smplayer. 
Failed after restart. 
Fails with either option selected for "prefs > network > support for video sites". 

My linux:
```
penguin:~/bin$ ls
yt-dlp
```

Is it ok if i made it executable after moving to bin?

Here's error:
/usr/bin/mpv --no-config --no-quiet --terminal --no-msg-color --input-ipc-server=/tmp/smplayer-mpv-427 --msg-level=ffmpeg/demuxer=error --no-fs --hwdec=no --sub-auto=fuzzy --no-input-default-bindings --input-vo-keyboard=no --no-input-cursor --cursor-autohide=no --no-keepaspect --wid=4194336 --monitorpixelaspect=1 --osd-level=1 --osd-scale=1 --osd-bar-align-y=0.6 --sub-ass --embeddedfonts --sub-ass-line-spacing=0 --sub-scale=1 --sub-font=Arial --sub-color=#ffffffff --sub-shadow-color=#ff000000 --sub-border-color=#ff000000 --sub-border-size=0.75 --sub-shadow-offset=2.5 --sub-font-size=50 --sub-bold=no --sub-italic=no --sub-margin-y=8 --sub-margin-x=20 --sub-codepage=ISO-8859-1 --sub-pos=100 --volume=55 --cache=auto --screenshot-template=cap_%F_%p_%02n --screenshot-format=jpg --screenshot-directory=/tmp/smplayer_screenshots --audio-pitch-correction=yes --volume-max=110 --ytdl --term-playing-msg=MPV_VERSION=${=mpv-version:}
INFO_VIDEO_WIDTH=${=width}
INFO_VIDEO_HEIGHT=${=height}
INFO_VIDEO_ASPECT=${=video-params/aspect}
INFO_VIDEO_FPS=${=container-fps:${=fps}}
INFO_VIDEO_FORMAT=${=video-format}
INFO_VIDEO_CODEC=${=video-codec}
INFO_AUDIO_FORMAT=${=audio-codec-name}
INFO_AUDIO_CODEC=${=audio-codec}
INFO_AUDIO_RATE=${=audio-params/samplerate}
INFO_AUDIO_NCH=${=audio-params/channel-count}
INFO_LENGTH=${=duration:${=length}}
INFO_DEMUXER=${=current-demuxer:${=demuxer}}
INFO_SEEKABLE=${=seekable}
INFO_TITLES=${=disc-titles}
INFO_CHAPTERS=${=chapters}
INFO_TRACKS_COUNT=${=track-list/count}
METADATA_TITLE=${metadata/by-key/title:}
METADATA_ARTIST=${metadata/by-key/artist:}
METADATA_ALBUM=${metadata/by-key/album:}
METADATA_GENRE=${metadata/by-key/genre:}
METADATA_DATE=${metadata/by-key/date:}
METADATA_TRACK=${metadata/by-key/track:}
METADATA_COPYRIGHT=${metadata/by-key/copyright:}
INFO_MEDIA_TITLE=${=media-title:}
INFO_STREAM_PATH=${stream-path}
 --audio-client-name=SMPlayer --term-status-msg=STATUS: ${=time-pos} / ${=duration:${=length:0}} P: ${=pause} B: ${=paused-for-cache} I: ${=core-idle} VB: ${=video-bitrate:0} AB: ${=audio-bitrate:0} https://www.youtube.com/watch?v=5ZdmS-EAbHo

[ytdl_hook] ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output. 
[ytdl_hook] youtube-dl failed: unexpected error ocurred 
[ytdl_hook] It appears that your youtube-dl version is severely out of date. 
Failed to recognize file format.
Exiting... (Errors when loading file)

**Comment by smplayer-dev** [comment]
Maybe it's still using the old version.
Type this in a console:
`yt-dlp --version`

The result should be 2023.07.06

**Comment by johnaweiss** [comment]
Thx for that.
I have 2023.07.06
